cms/models/pluginmodel.py

```diff
@@ -292,9 +292,8 @@ def copy_plugin(self, target_placeholder, target_language, parent_cache, no_sign
             new_plugin._no_reorder = True
         new_plugin.save()
         if plugin_instance:
-            if plugin_instance.__class__ == CMSPlugin:
-                # get a new instance so references do not get mixed up
-                plugin_instance = CMSPlugin.objects.get(pk=plugin_instance.pk)
+            # get a new instance so references do not get mixed up
+            plugin_instance = plugin_instance.__class__.objects.get(pk=plugin_instance.pk)
             plugin_instance.pk = new_plugin.pk
             plugin_instance.id = new_plugin.pk
```

Comment:
My understanding is that this is required to return a functional / correct object with right values (`plugin_instance` is returned and is later used by other functions) without hitting the database again as Django ORM does not make this magic for us.
It's ugly? Yes, I think so.
We should investigate this further for 3.1? I totally agree